File: FloodFillBasic.py
def dfs(img,x,y,o,n):
    if x<0 or x>=len(img) or y<0 or y>=len(img[0]) or img[x][y]!=o :
       return

    img[x][y]=n
    dfs(img,x+1,y,o,n)
    dfs(img,x-1,y,o,n)
    dfs(img,x,y+1,o,n)
    dfs(img,x,y-1,o,n)
    return img
    
def floodfill(img,sr,sc,newC):
    if img[sr][sc]==newC :
        return img
    oldC=img[sr][sc]
    dfs(img,sr,sc,oldC,newC)
    return img
    


img=[[1,1,1,0],[1,0,0,1],[1,1,1,1],[0,1,0,1]]
floodfill(img,0,0,3)

print(img)

File: FloodFillBasic.cpp
#include<bits/stdc++.h>
using namespace std;

void dfs(vector<vector<int>>&image,int x,int y,int oldC,int newC){
    if(x<0 || x>=image.size()||
       y<0 || y>=image[0].size()||
       image[x][y]!=oldC){
        return;
       }
       image[x][y]=newC;
       dfs(image,x+1,y,oldC,newC);
       dfs(image,x-1,y,oldC,newC);
       dfs(image,x,y+1,oldC,newC);
       dfs(image,x,y-1,oldC,newC);

      
}

vector<vector<int>> floodfill(vector<vector<int>>&image,int sr,int sc,int newColor){
    if(image[sr][sc]==newColor){
        return image;
    }
    int oldColor=image[sr][sc];
    dfs(image,sr,sc,oldColor,newColor);
    return image;
}
int main(){
    vector<vector<int>>image={
        {1,1,1},
        {1,1,0},
        {1,0,1}
    };
    floodfill(image,1,1,3);
    for(int i=0;i<image.size();i++){
        for(int j=0;j<image[0].size();j++){
            cout<<image[i][j]<<" ";
        
        }
        cout<<endl;
    }
}
